pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -365,14 +365,44 @@ func (cm *containerManagerImpl) NewPodContainerManager() PodContainerManager {
 			enforceCPULimits:  cm.EnforceCPULimits,
 			// cpuCFSQuotaPeriod is in microseconds. NodeConfig.CPUCFSQuotaPeriod is time.Duration (measured in nano seconds).
 			// Convert (cm.CPUCFSQuotaPeriod) [nanoseconds] / time.Microsecond (1000) to get cpuCFSQuotaPeriod in microseconds.
-			cpuCFSQuotaPeriod: uint64(cm.CPUCFSQuotaPeriod / time.Microsecond),
+			cpuCFSQuotaPeriod:   uint64(cm.CPUCFSQuotaPeriod / time.Microsecond),
+			podContainerManager: cm,
 		}
 	}
 	return &podContainerManagerNoop{
 		cgroupRoot: cm.cgroupRoot,
 	}
 }
 
+func (cm *containerManagerImpl) PodHasExclusiveCPUs(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.InitContainers {
+		exclusiveCPUs := cm.cpuManager.GetExclusiveCPUs(string(pod.UID), container.Name)
+		if !exclusiveCPUs.IsEmpty() {
+			klog.V(4).InfoS("Pod contains container with pinned cpus", "podName", pod.Name, "containerName", container.Name)
+			return true
+		}
+	}
+	for _, container := range pod.Spec.Containers {
+		exclusiveCPUs := cm.cpuManager.GetExclusiveCPUs(string(pod.UID), container.Name)
+		if !exclusiveCPUs.IsEmpty() {
+			klog.V(4).InfoS("Pod contains container with pinned cpus", "podName", pod.Name, "containerName", container.Name)
+			return true
+		}
+	}
```

Comment:
I'd reimplement this block just calling `ContainerHasExclusiveCPUs`

Reply:
done!